core/src/main/scala/org/apache/spark/sql/sources/StoreStrategy.scala

```diff
@@ -32,39 +32,52 @@ object StoreStrategy extends Strategy {
 
     case CreateTableUsing(tableIdent, userSpecifiedSchema, provider,
         false, opts, partitionColumns, bucketSpec, allowExisting, _) =>
-      ExecutedCommandExec(CreateMetastoreTableUsing(tableIdent, None,
+      SnappyExecutedCommandExec(CreateMetastoreTableUsing(tableIdent, None,
         userSpecifiedSchema, None, SnappyContext.getProvider(provider,
           onlyBuiltIn = false), allowExisting, opts, isBuiltIn = false)) :: Nil
+    case a@CreateTableUsingAsSelect(tableIdent, provider, partitionCols,
+      bucketSpec, mode, opts, _) =>
+      import scala.reflect.runtime.{universe => ru}
+      import ru._
+      val rm = runtimeMirror(this.getClass.getClassLoader)
+      val pr = rm.reflect(a)
+      // 2.0.0 parameter name: child
+      // 2.0.2 parameter name: query
+      // Use reflection to find the value of parameter
+      val childterm = typeOf[CreateTableUsingAsSelect].members.find(a =>
+        a.fullName.contains("query") || a.fullName.contains("child"))
+      val value = pr.reflectField(childterm.get.asTerm).get
```

Comment:
Instead of reflection a cleaner way would be using Product.productElement (see the patch I had sent):
```scala
    case c@CreateTableUsingAsSelect(tableIdent, provider, partitionCols,
    _, mode, opts, _) =>
      val query = c.productElement(6).asInstanceOf[LogicalPlan]
      // CreateTableUsingSelect is only invoked by DataFrameWriter etc
      // so that should support both builtin and external tables
      ExecuteCommand(CreateMetastoreTableUsingSelect(tableIdent, None,
        None, None, SnappyContext.getProvider(provider, onlyBuiltIn = false),
        temporary = false, partitionCols, mode, opts, query,
        isBuiltIn = false)) :: Nil
```